src/hashdump.c

```diff
@@ -102,36 +97,33 @@ int hashdump_main(int argc, char **argv)
 	}
 
 	if (!set_path_prefix(NULL)) {
-		free(file);
 		return EINIT_GLOBALS;
 	}
 
 	file.filename = strdup_or_die(argv[optind]);
 	// Accept relative paths if no path_prefix set on command line
 	if (use_prefix) {
-		fullname = mk_full_filename(path_prefix, file->filename);
+		fullname = mk_full_filename(path_prefix, file.filename);
 	} else {
-		fullname = strdup_or_die(file->filename);
+		fullname = strdup_or_die(file.filename);
 	}
 
 	fprintf(stderr, "Calculating hash %s xattrs for: %s\n",
-		(file->use_xattrs ? "with" : "without"), fullname);
+		(file.use_xattrs ? "with" : "without"), fullname);
 
-	populate_file_struct(file, fullname);
-	ret = compute_hash(file, fullname);
+	populate_file_struct(&file, fullname);
+	ret = compute_hash(&file, fullname);
 	if (ret != 0) {
 		fprintf(stderr, "compute_hash() failed\n");
 	} else {
-		printf("%s\n", file->hash);
-		if (file->is_dir) {
-			if (is_directory_mounted(fullname)) {
-				fprintf(stderr, "!! dumped hash might not match a manifest hash because a mount is active\n");
-			}
+		printf("%s\n", file.hash);
+		if (file.is_dir && is_directory_mounted(fullname)) {
+			fprintf(stderr, "!! dumped hash might not match a manifest"
+					"hash because a mount is active\n");
```

Comment:
I think this will print: "manifesthash" instead of "manifest hash"

Reply:
thanks. fixed.